plugin/inc/utils/toolbox.class.php

```diff
@@ -306,4 +306,23 @@ public static function getCodmatValue(string $codmat): string
         return self::$codmatValues[$codmat]['denum'] ?? '';
     }
 
+    public static function getIdentifierAttributeByAttribute(string $modelName, string $attributeValue, string $attributeToSearchFor = null, string $attributeToReturn = 'id'): mixed
+    {
+        if (!($table = getTableForItemType($modelName))) {
+            return null;
+        }
+
+        if (empty($attributeToSearchFor)) {
+            $attributeToSearchFor = $modelName::getNameField();
+        }
+
+        $queryResult = PluginIserviceDB::getQueryResult("SELECT $attributeToReturn FROM $table WHERE $attributeToSearchFor = '$attributeValue'");
+
+        if (count($queryResult) > 1) {
```

Comment:
what if there is no result?
```suggestion
        if (count($queryResult) != 1) {
```